### PR TITLE
python3Packages.openaiauth: init at 0.3.2

### PR DESCRIPTION
**Pull request by i-am-logger**
###### Description of changes

new package

###### Things done

A Python library for authenticating with the OpenAI API.

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/openaiauth/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+python3.pkgs.buildPythonPackage rec {
+  pname = "OpenAIAuth";
+  version = "0.3.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "CPcBgGvxRO677EdPI3lNtJXkCW7el6N6N2GeaDo5ApU=";
+  };
+
+  buildInputs = [ python3.pkgs.setuptools ];
```
Comment (NobbZ) [inline pkgs/development/python-modules/openaiauth/default.nix:12]: Please use `setuptools` and also add it to the argset.

File: pkgs/development/python-modules/openaiauth/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ fetchPypi, python3, lib }:
+
+python3.pkgs.buildPythonPackage rec {
```
Comment (NobbZ) [inline pkgs/development/python-modules/openaiauth/default.nix:3]: Please use `buildPythonPackage` directly and add it to the argset.

File: pkgs/development/python-modules/openaiauth/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  version = "0.3.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "CPcBgGvxRO677EdPI3lNtJXkCW7el6N6N2GeaDo5ApU=";
+  };
+
+  buildInputs = [ python3.pkgs.setuptools ];
+
+  propagatedBuildInputs = [ python3.pkgs.requests ];
```
Comment (NobbZ) [inline pkgs/development/python-modules/openaiauth/default.nix:14]: Just `request` and add to the argset above.

**Review by NobbZ — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/openaiauth/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{ buildPythonPackage, fetchPypi, setuptools, requests, lib }:
+
+buildPythonPackage rec {
+  pname = "OpenAIAuth";
+  version = "0.3.2";
+
```
Comment (NobbZ) [inline pkgs/development/python-modules/openaiauth/default.nix:6]:
```suggestion

  format = "setuptools";

```

File: pkgs/development/python-modules/openaiauth/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  buildInputs = [ setuptools ];
+
+  propagatedBuildInputs = [ requests ];
+
+  meta = with lib; {
+    description = "A Python library for authenticating with the OpenAI API.";
+    license = licenses.mit;
+    maintainers = with maintainers; [ realsnick ];
+    homepage = "https://github.com/openai/openai-python/tree/master/openai/auth";
```
Comment (NobbZ) [inline pkgs/development/python-modules/openaiauth/default.nix:20]:
```suggestion
    homepage = "https://github.com/acheong08/OpenAIAuth";
```

The original link 404s for me, pypi says this link was correct.

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -18310,6 +18310,8 @@ with pkgs;
 
   openai = with python3Packages; toPythonApplication openai;
 
+  openaiauth = with python3.pkgs; toPythonApplication openaiauth;
```
Comment (NobbZ) [inline pkgs/top-level/all-packages.nix:18313]: Please remove this, the package does not include any runnable, it is a pure library package.

**Review by NobbZ — CHANGES_REQUESTED** [review]
Please also reword your commits message to:

```
python3Packages.openaiauth: init at 0.3.2
```

File: pkgs/top-level/all-packages.nix (hunk 6 above)
Comment (realsnick) [reply pkgs/top-level/all-packages.nix]: done

**Comment by NobbZ** [comment]
You missed one thing:

https://github.com/NixOS/nixpkgs/pull/217725#pullrequestreview-1319215398

**Review by NobbZ — APPROVED** [review]

**Comment by fabaff** [comment]
As mentioned in https://github.com/NixOS/nixpkgs/pull/217725#pullrequestreview-1319215398 the commit message should be `python310Packages.openaiauth: init at 0.3.2`

**Comment by NobbZ** [comment]
> As mentioned in [#217725 (review)](https://github.com/NixOS/nixpkgs/pull/217725#pullrequestreview-1319215398) the commit message should be `python310Packages.openaiauth: init at 0.3.2`

I was pretty sure that was the case at the time of my approval…

**Review by NobbZ — CHANGES_REQUESTED** [review]
Commit message as mentioned above

**Review by NobbZ — APPROVED** [review]

File: pkgs/development/python-modules/openaiauth/default.nix [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "CPcBgGvxRO677EdPI3lNtJXkCW7el6N6N2GeaDo5ApU=";
+  };
+
+  buildInputs = [ setuptools ];
+
+  propagatedBuildInputs = [ requests ];
+
```
[the remark this answers is not in the document]
Comment (realsnick) [reply pkgs/development/python-modules/openaiauth/default.nix:17]: Thanks for the suggestions, will apply all shortly

**Review by NobbZ — APPROVED** [review]

**Pull request merged**